lib/_emerge/EbuildPhase.py

```diff
@@ -130,9 +131,21 @@ def _start(self):
 
 		if self.phase == 'package':
 			if 'PORTAGE_BINPKG_TMPFILE' not in self.settings:
-				self.settings['PORTAGE_BINPKG_TMPFILE'] = \
-					os.path.join(self.settings['PKGDIR'],
-					self.settings['CATEGORY'], self.settings['PF']) + '.tbz2'
+				binpkg_format = self.settings.get("BINPKG_FORMAT", "xpak")
+				if binpkg_format == "xpak":
+					self.settings['BINPKG_FORMAT'] = "xpak"
+					self.settings['PORTAGE_BINPKG_TMPFILE'] = \
+						os.path.join(self.settings['PKGDIR'],
+						self.settings['CATEGORY'], self.settings['PF']) \
+						+ '.tbz2'
+				elif binpkg_format == "gpkg":
+					self.settings['BINPKG_FORMAT'] = "gpkg"
```

Comment:
Isn't this redundant?

Reply:
yes, assume the BINPKG_FORMAT may not actually exists, but from default value.